### PR TITLE
Trainer schedule

### PR DESCRIPTION
**Pull request by aravindhan-converse**
Trainer & Schedule module

### CODE REVIEW

File: app/controllers/admin/trainer_courses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          ]
+        end
+        format.json { render json: @trainer.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  private
+
+  def render_invalid_trainer(format)
```
Comment (aravintsamy-converse) [inline app/controllers/admin/trainer_courses_controller.rb]: find the use or else remove it

File: app/models/trainer_course.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class TrainerCourse < ApplicationRecord
+  belongs_to :trainer
+  belongs_to :course
+  has_many :trainer_attendances
+
+  # validates :trainer_id, presence: true
```
Comment (aravintsamy-converse) [inline app/models/trainer_course.rb]: remove it

**Review by aravintsamy-converse — APPROVED** [review]

File: app/controllers/admin/batches_controller.rb [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+          ]
+        end
+        format.json { render json: @batch.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def show
+    respond_to do |format|
+      if @batch
```
Comment (sivakumar-converse) [inline app/controllers/admin/batches_controller.rb:37]: alignment not correct

File: app/controllers/admin/batches_controller.rb [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            turbo_stream.append('batch-table', partial: 'shared/failed', locals: { message: 'Batch not found.', type: 'notice' })
+          ]
+        end
+        format.json { render json: @batch.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def edit
+    render layout: false
```
Comment (sivakumar-converse) [inline app/controllers/admin/batches_controller.rb:52]: use in class

File: app/controllers/admin/batches_controller.rb [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+      format.turbo_stream
+      format.json { render json: @batches }
+    end
+  end
+
+  def create
+    @batch = Batch.new(batch_params)    
+    respond_to do |format|
+      if @batch.save   
+        @batches = Batch.get_batches(params)
```
Comment (sivakumar-converse) [inline app/controllers/admin/batches_controller.rb:21]: use around action

File: app/controllers/admin/batches_controller.rb [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                                                                  locals: { message: 'Batch deletion failed.', type: 'notice' })
+        end
+      end
+    end
+  end
+
+  def find_course_trainers
+    respond_to do |format|
+      if @course
+          @trainers = @course.trainers
```
Comment (sivakumar-converse) [inline app/controllers/admin/batches_controller.rb:100]: use lint

File: app/controllers/admin/batches_controller.rb [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+          format.turbo_stream do
+            render turbo_stream: turbo_stream.append('batch-table', partial: 'shared/failed', locals: { message: 'Course not found.', type: 'notice' })
+          end
+      end
+    end
+  end
+
+  def edit_find_course_trainers
+    respond_to do |format|
+      if @course
```
Comment (sivakumar-converse) [inline app/controllers/admin/batches_controller.rb:112]: use lint

**Review by sivakumar-converse — CHANGES_REQUESTED** [review]

File: app/controllers/admin/trainer_courses_controller.rb (hunk 1 above)
Comment (aravindhan-converse) [reply app/controllers/admin/trainer_courses_controller.rb]: removed

File: app/models/trainer_course.rb (hunk 2 above)
Comment (aravindhan-converse) [reply app/models/trainer_course.rb]: removed

File: app/controllers/admin/batches_controller.rb (hunk 3 above)
Comment (aravindhan-converse) [reply app/controllers/admin/batches_controller.rb:37]: fixed

File: app/controllers/admin/batches_controller.rb (hunk 6 above)
Comment (aravindhan-converse) [reply app/controllers/admin/batches_controller.rb:100]: fixed

File: app/controllers/admin/batches_controller.rb (hunk 7 above)
Comment (aravindhan-converse) [reply app/controllers/admin/batches_controller.rb:112]: fixed

**Review by sivakumar-converse — APPROVED** [review]

**Pull request merged**